cmd/cagent/main.go

```diff
@@ -28,99 +28,81 @@ var (
 
 const defaultLogLevel = "error"
 
-func askForConfirmation(s string) bool {
-	reader := bufio.NewReader(os.Stdin)
-
-	for {
-		fmt.Printf("%s [y/n]: ", s)
-
-		response, err := reader.ReadString('\n')
-		if err != nil {
-			log.Fatal(err)
-		}
-
-		response = strings.ToLower(strings.TrimSpace(response))
-
-		if response == "y" || response == "yes" {
-			return true
-		} else if response == "n" || response == "no" {
-			return false
-		}
-	}
-}
-
-func main() {
-	ca := cagent.New()
-	ca.SetVersion(version)
-
-	sigc := make(chan os.Signal, 1)
-	signal.Notify(sigc,
-		syscall.SIGHUP,
-		syscall.SIGINT,
-		syscall.SIGTERM)
-
-	var serviceInstallUserPtr *string
-	var serviceInstallPtr *bool
-	outputFilePtr := flag.String("o", "", "file to write the results")
-
-	cfgPathPtr := flag.String("c", cagent.DefaultCfgPath, "config file path")
-	logLevelPtr := flag.String("v", defaultLogLevel, "log level – overrides the level in config file (values \"error\",\"info\",\"debug\")")
-	systemManager := service.ChosenSystem()
-	daemonizeModePtr := flag.Bool("d", false, "daemonize – run the proccess in background")
-	oneRunOnlyModePtr := flag.Bool("r", false, "one run only – perform checks once and exit. Overwrites output file")
+var ca *cagent.Cagent
+var systemManager service.System
+
+func init() {
```

Comment:
Why have you moved so much logic into the `init` function?
I don't think things like reading config file or dumping config or setting a logger belongs in this function.